code/_onclick/hud/plane_master_controller.dm

```diff
@@ -13,8 +13,6 @@ INITIALIZE_IMMEDIATE(/atom/movable/plane_master_controller)
 	if(!istype(hud))
 		return
 	owner_hud = hud
-	if(istype(owner_hud))
```

Comment:
I think i somehow missed the ! In here thats the issue this originally had the intention of checking a specific state that could happen during a unit test and delete it in this case trough it should not cause a harddelete cause this still applies the normal rules of qdelete and calls the destroy proc in the end as usual

Reply:
But yes this should not delete directly after initialize the current state is completly wrong here

Reply:
The only reason this probably causes a harddelete is cause it never removes the reference in the variable of the owner

Reply:
Maybe instead of this check blacklist the typpath in code/modules/unit_tests/create_and_destroy.dm just add it into the ignore list there

Reply:
That will keep it from beeing run by the unit check while not interfering with the normal initialize proc

Reply:
Set to be ignored in create_and_destroy